server/chat-plugins/mafia.ts

```diff
@@ -2679,11 +2695,16 @@ export const commands: ChatCommands = {
 			}
 			this.room = targetRoom;
 			const game = this.requireGame(MafiaTracker);
-			if (game.hostid !== user.id && !game.cohosts.includes(user.id)) this.checkCan('mute', null, targetRoom);
+			if (game.hostid !== user.id && !game.cohostids.includes(user.id)) this.checkCan('mute', null, targetRoom);
 			if (!toID(args.join(''))) return this.parse('/help mafia revive');
 			let didSomething = false;
 			for (const targetUsername of args) {
-				if (game.revive(user, toID(targetUsername), cmd === 'forceadd')) didSomething = true;
+				const targetUser = Users.get(toID(targetUsername));
+				if (targetUser && MafiaTracker.isGameBanned(targetUser)) {
```

Comment:
this check (and the one in join) should probably be moved into `canJoin`

Reply:
also is it intentional that gamebans are global, while hostbans only apply in the mafia room?